packages/react-native-reanimated/src/logger/logger.ts

```diff
@@ -55,27 +66,59 @@ export function logToLogBoxAndConsole(data: LogData) {
 }
 
 /**
- * Replaces the default log function with a custom implementation.
+ * Updates logger configuration.
  *
- * @param logFunction - The custom log function.
+ * @param config - The new logger configuration to apply.
+ *   - level: The minimum log level to display.
+ *   - strict: Whether to log warnings and errors that are not strict.
+ *    Defaults to false.
+ *   - logFunction: The function to use for logging.
  */
-export function replaceLoggerImplementation(
-  logFunction: (data: LogData) => void
+export function updateLoggerConfig(config?: Partial<LoggerConfigInternal>) {
+  'worklet';
+  global.__loggerConfig = {
+    logFunction:
+      // Re-use previously assigned log function if it exists
+      config?.logFunction ?? global.__loggerConfig?.logFunction ?? logToConsole,
+    // Don't reuse previous level and strict values from the global config
+    level: config?.level ?? 'warn',
+    strict: config?.strict ?? false,
+  };
+}
+
+const logLevelImportance = {
+  warn: 1,
+  error: 2,
+  fatal: 3,
+} as const;
+
+function handleLog(
+  level: Exclude<LogLevel, 'syntax'>,
+  message: string,
+  options: LogOptions
 ) {
-  loggerImpl.logFunction = logFunction;
+  'worklet';
+  const config = global.__loggerConfig;
+  if (
+    (options.strict && !config.strict) ||
+    logLevelImportance[level] < logLevelImportance[config.level]
+  ) {
+    return;
+  }
+  config.logFunction(createLog(level, message));
 }
 
 export const logger = {
-  warn(message: string) {
+  warn(message: string, options: LogOptions = {}) {
     'worklet';
-    loggerImpl.logFunction(createLog('warn', message));
+    handleLog('warn', message, options);
```

Comment:
I assume the options parameter got added to allow for dynamically silencing/escalating warnings etc., is it correct?

Reply:
currently it is only used to mark specific logs as **strict-only**. I don't know which logs will be suitable to convert to strict mode only but probably these that are most annoying (like overriding layout animation with animated style which is often done on purpose or my [PR](https://github.com/software-mansion/react-native-reanimated/pull/6310) that should be merged with invalid shared value read/write).

See `LogOptions` type, currently it can take just one option.
```tsx
type LogOptions = {
  strict?: boolean;
};
```